search-resources-api/src/main/java/no/unit/nva/search/ElasticSearchHighLevelRestClient.java

```diff
@@ -0,0 +1,201 @@
+        elasticSearchEndpointAddress = environment.readEnv(ELASTICSEARCH_ENDPOINT_ADDRESS_KEY);
+        elasticSearchEndpointIndex = environment.readEnv(ELASTICSEARCH_ENDPOINT_INDEX_KEY);
+        elasticSearchRegion = environment.readEnv(ELASTICSEARCH_ENDPOINT_REGION_KEY);
+        elasticSearchClient = createElasticsearchClientWithInterceptor(SERVICE_NAME,
+                elasticSearchRegion,
+                elasticSearchEndpointAddress);
+        logger.info(INITIAL_LOG_MESSAGE, elasticSearchEndpointAddress, elasticSearchEndpointIndex);
+    }
+
+    /**
+     * Creates a new ElasticSearchRestClient.
+     *
+     * @param environment Environment with properties
+     * @param elasticSearchClient client to use for access to ElasticSearch
+     */
+    public ElasticSearchHighLevelRestClient(Environment environment, RestHighLevelClient elasticSearchClient) {
+        elasticSearchEndpointAddress = environment.readEnv(ELASTICSEARCH_ENDPOINT_ADDRESS_KEY);
+        elasticSearchEndpointIndex = environment.readEnv(ELASTICSEARCH_ENDPOINT_INDEX_KEY);
+        elasticSearchRegion = environment.readEnv(ELASTICSEARCH_ENDPOINT_REGION_KEY);
+        this.elasticSearchClient = elasticSearchClient;
+        logger.info(INITIAL_LOG_MESSAGE, elasticSearchEndpointAddress, elasticSearchEndpointIndex);
+    }
+
+    /**
+     * Searches for an term or index:term in elasticsearch index.
+     * @param term search argument
+     * @param results number of results
+     * @throws ApiGatewayException thrown when uri is misconfigured, service i not available or interrupted
+     */
+    public SearchResourcesResponse searchSingleTerm(String term, int results) throws ApiGatewayException {
+        try {
+            QueryStringQueryBuilder queryBuilder = QueryBuilders.queryStringQuery(term);
+            final SearchSourceBuilder sourceBuilder = new SearchSourceBuilder()
+                .query(queryBuilder)
+                .size(results);
+            final SearchRequest searchRequest = new SearchRequest(elasticSearchEndpointIndex);
+            searchRequest.source(sourceBuilder);
+            logger.debug(SEARCH_REQUEST_MESSAGE, elasticSearchEndpointIndex, term, searchRequest);
+            SearchResponse searchResponse = elasticSearchClient.search(searchRequest, RequestOptions.DEFAULT);
+            return toSearchResourcesResponse(searchResponse.toString());
+        } catch (Exception e) {
+            throw new SearchException(e.getMessage(), e);
+        }
+    }
+
+    /**
+     * Adds or insert a document to an elasticsearch index.
+     * @param document the document to be inserted
+     * @throws SearchException when something goes wrong
+     * */
+    public void addDocumentToIndex(IndexDocument document) throws SearchException {
+
+        logger.debug(UPSERTING_LOG_MESSAGE, document);
+
+        try {
+            String jsonDocument = document.toJsonString();
+            IndexRequest indexRequest = new IndexRequest(elasticSearchEndpointIndex)
```

Comment:
Possible to extract 133-137 to doUpsert method?

Reply:
Extracted